tokenserver/views.py

```diff
@@ -151,29 +151,31 @@ def _valid_browserid_assertion(request, assertion):
     request.metrics['token.assertion.verify_success'] = 1
     request.validated['authorization'] = assertion
 
+    email = assertion['email']
+    request.validated['fxa_uid'] = email.split("@", 1)[0]
+
     id_key = request.registry.settings.get("fxa.metrics_uid_secret_key")
     if id_key is None:
         id_key = 'insecure'
-    email = assertion['email']
-    fxa_uid_full = fxa_metrics_hash(email, id_key)
+    hashed_fxa_uid_full = fxa_metrics_hash(email, id_key)
     # "legacy" key used by heka active_counts.lua
-    request.metrics['uid'] = fxa_uid_full
+    request.metrics['uid'] = hashed_fxa_uid_full
     request.metrics['email'] = email
 
     # "new" keys use shorter values
-    fxa_uid = fxa_uid_full[:32]
-    request.validated['fxa_uid'] = fxa_uid
-    request.metrics['fxa_uid'] = fxa_uid
+    hashed_fxa_uid = hashed_fxa_uid_full[:32]
+    request.validated['hashed_fxa_uid'] = hashed_fxa_uid
+    request.metrics['hashed_fxa_uid'] = hashed_fxa_uid
```

Comment:
Just bike-shedding this property name for a moment (sorry), but I quite like how your explanatory comment calls this a hashed metrics uid, it just makes the distinction a bit clearer. Would `hashed_metrics_uid` be a more helpful name here?

Reply:
(or even just `metrics_uid` as the hashing seems like an implementation detail)

Reply:
Bikeshedding very welcome, although we'll need to take care not to break any consumers of the existing names (e.g. changing the key used for `request.metrics` might not be possible without also updating some downstream lua filters).  I think including "metrics" in the name explicitly makes sense, esp. since the main reason we do it this way is for amplitude.